tests/sample/map_reuse.c

```diff
@@ -19,7 +19,7 @@
 #include "bpf_helpers.h"
 
 #define PIN_NONE 0
-#define PIN_GLOBAL_NS 2
+#define PIN_BY_NAME 1
```

Comment:
line 19 incudes bpf_helpers.h which includes libbpf/src/bpf_helpers.h which defines LIBBPF_PIN_BY_NAME.
So there's no reason to define something as "1".   Instead change the program to use LIBBPF_PIN_BY_NAME, or at least define PIN_BY_NAME to be LIBBPF_PIN_BY_NAME.

Reply:
PR #2880 is updating bp_helpers.h so that we no longer include `libbpf/src/bpf_helpers.h`.

As with the other changes in PR #2880, we can redefine LIBBPF_PIN_BY_NAME in bpf_helpers.h and then define PIN_BY_NAME to LIBBPF_PIN_BY_NAME?

Reply:
I don't see any reason to use "PIN_BY_NAME" as a macro, just use LIBBPF_PIN_BY_NAME everywhere.

Reply:
Currently a lot of code in user mode uses the pin type enum `ebpf_pin_type_t`. If we define PIN_BY_NAME to LIBBPF_PIN_BY_NAME, then we can use LIBBPF_PIN_BY_NAME everywhere.

Or, maybe we get rid of the `ebpf_pin_type_t` enum completely and typedef ebpf_pin_type_t to say uint32_t? Then we dont need PIN_BY_NAME.

    `typedef ebpf_pin_type_t uint32_t`

Reply:
Updated enum values to match the libbpf names. Since `libbpf/src/bpf_helpers.h` is not included anymore, we cannot use the definition from that header file, and need to redefine it.